labonneboite/common/pro.py

```diff
@@ -9,6 +10,28 @@
 
 PRO_VERSION_SESSION_KEY = 'pro_version'
 
+
+def ips_from_ip_ranges(ip_ranges):
+    """
+    Return a list of IP addresses from a list of IP ranges.
+
+    Usage example:
+        ips_from_ip_ranges(['192.168.0.0', '192.168.0.0/22'])
+
+    Input:
+        ip_ranges: list of strings
+
+    Output:
+        list of ipaddress.IPv4Address objects
+
+    """
+    return [
+        ip for ip_range in ip_ranges
+        for ip in ipaddress.IPv4Network(ip_range)
+    ]
```

Comment:
wow ça me fait un noeud au cerveau ça 

perso j'avoue que je trouve `[x for x in a for a in b]` bien plus lisible que `[x for a in b for x in a]` car cela se lit directement en langue anglaise

mais apparemment TIL j'ai tort et comme toi la plupart des gens utilisent `[x for a in b for x in a]`

source https://stackoverflow.com/questions/17657720/python-list-comprehension-double-for/17657966

je pense que cet ordre vient de 

```
for a in b:
    for x in a:
        x
```

Reply:
Oui je suis d'accord, ce n'est pas hyper compréhensible (ce qui est fort dommage, quand même, pour une compréhension de liste !). Je vais voir comment la modifier pour qu'elle soit plus claire.